examples/vpc_service_control/onprem_project/main.tf

```diff
@@ -0,0 +1,285 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+provider "google" {
+  credentials = "${file("${var.service_account_file}")}"
+}
+
+resource "google_project" "on_prem_network_project" {
+  name = "On Prem Network"
+  project_id = "${var.project_id}"
+  org_id     = "${var.organization_id}"
+  billing_account	= "${var.billing_account_id}"
+  auto_create_network = true
+}
+
+
+/* STEP 1A: DELETE THIS COMMENT LINE TO RESERVE STATIC IP (costs ~$0.24/day) 
+resource "google_compute_address" "onprem_vpn_ip" {
+  address_type = "EXTERNAL"
+  name         = "onprem-vpn-ip"
+  network_tier = "PREMIUM"
+  project      = "${google_project.on_prem_network_project.project_id}"
+  region       = "${var.cloud_router_region}"
+} 
+
+output "ip_addr_of_onprem_vpn_router" {
+  value = "${google_compute_address.onprem_vpn_ip.address}"
+}
+STEP 1B: DELETE THIS END-COMMENT LINE.  Then run terraform apply. */
+
+
+/* STEP 3A: DELETE THIS COMMENT LINE TO CREATE ONPREM VPN ROUTER (costs ~$1.80/day)
+resource "google_compute_router" "onprem_cloud_router" {
+  bgp {
+    advertise_mode = "DEFAULT"
+    asn            = "64512"
+  }
+
+  name    = "onprem-cloud-router"
+  network = "default"
+  project = "${google_project.on_prem_network_project.project_id}"
+  region  = "${var.cloud_router_region}"
+}
+
+resource "google_compute_vpn_gateway" "target_gateway" {
+  name    = "target-vpn-gateway"
+  network = "default"
+  project = "${google_project.on_prem_network_project.project_id}"
+  region  = "${var.cloud_router_region}"
+}
+
+resource "google_compute_forwarding_rule" "fr_for_vpn_gateway" {
+  name        = "frforvpngateway"
+  ip_protocol = "ESP"
+  ip_address  = "${google_compute_address.onprem_vpn_ip.address}"
+  target      = "${google_compute_vpn_gateway.target_gateway.self_link}"
+  project = "${google_project.on_prem_network_project.project_id}"
+  region = "${var.cloud_router_region}"
+}
+
+resource "google_compute_forwarding_rule" "fr_udp500" {
+  name        = "frforvpngatewayudp500"
+  ip_protocol = "UDP"
+  port_range = "500"
+  ip_address  = "${google_compute_address.onprem_vpn_ip.address}"
+  target      = "${google_compute_vpn_gateway.target_gateway.self_link}"
+  project = "${google_project.on_prem_network_project.project_id}"
+  region = "${var.cloud_router_region}"
+}
+
+resource "google_compute_forwarding_rule" "fr_udp4500" {
+  name        = "frforvpngatewayudp4500"
+  ip_protocol = "UDP"
+  port_range = "4500"
+  ip_address  = "${google_compute_address.onprem_vpn_ip.address}"
+  target      = "${google_compute_vpn_gateway.target_gateway.self_link}"
+  project = "${google_project.on_prem_network_project.project_id}"
+  region = "${var.cloud_router_region}"
+}
+
+resource "google_compute_vpn_tunnel" "onprem_vpn_tunnel" {
+  ike_version             = "2"
+  name                    = "onprem-vpn-tunnel"
+  peer_ip                 = "${var.ip_addr_of_cloud_vpn_router}"
```

Comment:
```suggestion
  peer_ip                 = "${google_compute_address.vpc_sc_vpn_ip.address}"
```

Reply:
This suggested change causes an error, because `vpc_sc_vpn_ip` does not exist in this main.tf file; it exists in the other main.tf file for the VPC SC project.

`Error: resource 'google_compute_vpn_tunnel.onprem_vpn_tunnel' config: unknown resource 'google_compute_address.vpc_sc_vpn_ip' referenced in variable google_compute_address.vpc_sc_vpn_ip.address`

Reply:
Indeed. I should've added a comment saying this suggestion is assuming both VPNs are created in the same root config and won't work with your current setup.